test/Chainweb/Test/Utils.hs

```diff
@@ -280,24 +281,29 @@ testRocksDb
     :: B.ByteString
     -> RocksDb
     -> IO RocksDb
-testRocksDb l = rocksDbNamespace (const prefix)
-  where
-    prefix = (<>) l . sshow <$> (randomIO @Word64)
+testRocksDb l r = do
+  prefix <- (<>) l . sshow <$> (randomIO @Word64)
+  return r { _rocksDbNamespace = prefix }
 
 withRocksResource :: (IO RocksDb -> TestTree) -> TestTree
 withRocksResource m = withResource create destroy wrap
   where
     create = do
       sysdir <- getCanonicalTemporaryDirectory
       dir <- createTempDirectory sysdir "chainweb-rocksdb-tmp"
-      rocks <- openRocksDb dir
-      return (dir, rocks)
-    destroy (dir, rocks) = do
+      opts@(R.Options' opts_ptr _ _) <- R.mkOpts modernDefaultOptions
+      rocks <- openRocksDb dir opts_ptr
+      return (dir, rocks, opts)
+    destroy (dir, rocks, opts) = do
         closeRocksDb rocks
+          `catchAllSynchronous` (const $ return ())
```

Comment:
Could this possibly hide exception that we'd like to catch during testing? If there are expected exceptions, maybe we could be more specific in what we catch?

Reply:
Replacing with `finally`.